### PR TITLE
Added ROS 2 sensor debugger.

### PR DESCRIPTION
**Pull request by michalpelka**
The simple gem that presents how to adjust ROS 2 sensors in runtime.
image

### CODE REVIEW

File: Gems/SensorDebug/Code/Include/SensorDebug/SensorDebugTypeIds.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace SensorDebug
+{
+    // System Component TypeIds
+    inline constexpr const char* SensorDebugSystemComponentTypeId = "{079B7923-E178-48AA-A0BD-9908BDB9F2BB}";
+    inline constexpr const char* SensorDebugEditorSystemComponentTypeId = "{EEB09214-4772-4B21-8CF5-F29A9FF3CE1C}";
+
+    // Module derived classes TypeIds
+    inline constexpr const char* SensorDebugModuleInterfaceTypeId = "{6273F4BC-9B69-4567-A748-B602FBFCB6CE}";
+    inline constexpr const char* SensorDebugModuleTypeId = "{28353925-BFF1-49A2-B003-5FC22530672D}";
```
Comment (patrykantosz) [inline Gems/SensorDebug/Code/Include/SensorDebug/SensorDebugTypeIds.h:12]:
```suggestion
    inline constexpr const char* SensorDebugModuleTypeId = "{28353925-BFF1-49A2-B003-5FC22530672D}";

```

File: Gems/SensorDebug/Code/Source/Clients/SensorDebugSystemComponent.cpp [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+                    m_sensorNames[pair] = component->RTTI_GetTypeName();
+                    m_entitiesNames[pair] = entity->GetName();
+
+                    ROS2::SensorConfiguration sensorConfig = handler->GetSensorConfiguration();
+                    m_sensorFrequencies[pair] = sensorConfig.m_frequency;
+                }
+                return true;
+            });
+
+        AZ_Printf("test", "Found %d sensor entities", m_sensorEntities.size());
```
Comment (patrykantosz) [inline Gems/SensorDebug/Code/Source/Clients/SensorDebugSystemComponent.cpp:67]:
```suggestion
        AZ_Printf("TestComponent", "Found %d sensor entities", m_sensorEntities.size());
```

File: Gems/SensorDebug/Code/Source/Clients/SensorDebugSystemComponent.cpp [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        AZ_Printf("TestComponent", "Found %d sensor entities", m_sensorEntities.size());
+    }
+
+    void SensorDebugSystemComponent::FindSensorsWithGivenType(const char* typeId)
+    {
+        ClearSensors();
+        auto searchFunction = [this, typeId](AZ::Entity* entity)
+        {
+            AZ_Assert(entity, "Entity is null");
+            auto sensorComponent = ROS2::GetSensorsOfType(entity->GetId(), AZ::Uuid(typeId));
```
Comment (patrykantosz) [inline Gems/SensorDebug/Code/Source/Clients/SensorDebugSystemComponent.cpp:106]:
```suggestion
            const auto sensorComponent = ROS2::GetSensorsOfType(entity->GetId(), AZ::Uuid(typeId));
```

File: Gems/SensorDebug/Code/Source/Clients/SensorDebugSystemComponent.cpp [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+            if (ImGui::Button(buttonSetFreq.c_str()))
+            {
+                ROS2::SensorConfigurationRequestBus::Event(
+                    sensorEntity, &ROS2::SensorConfigurationRequest::SetDesiredFrequency, m_sensorFrequencies[sensorEntity]);
+            }
+        }
+
+        ImGui::End();
+    }
+
```
Comment (patrykantosz) [inline Gems/SensorDebug/Code/Source/Clients/SensorDebugSystemComponent.cpp:246]:
```suggestion
```

File: Gems/SensorDebug/Code/Source/Clients/SensorDebugSystemComponent.h [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    private:
+        void FindSensorsWithBusAPI();
+        void FindSensorsWithComponentAPI();
+        void FindSensorsWithGivenType(const char* typeId);
+        AZStd::vector<AZ::EntityComponentIdPair> m_sensorEntities;
+        AZStd::unordered_map<AZ::EntityComponentIdPair, AZStd::string> m_sensorNames;
+        AZStd::unordered_map<AZ::EntityComponentIdPair, AZStd::string> m_entitiesNames;
+        AZStd::unordered_map<AZ::EntityComponentIdPair, float> m_sensorFrequencies;
+    };
+
```
Comment (patrykantosz) [inline Gems/SensorDebug/Code/Source/Clients/SensorDebugSystemComponent.h:45]:
```suggestion
```

File: Gems/SensorDebug/Code/Source/Tools/SensorDebugEditorSystemComponent.cpp [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    void SensorDebugEditorSystemComponent::OnStartPlayInEditorBegin()
+    {
+        ClearSensors();
+    }
+
+    void SensorDebugEditorSystemComponent::OnStopPlayInEditor()
+    {
+        ClearSensors();
+    }
+
```
Comment (patrykantosz) [inline Gems/SensorDebug/Code/Source/Tools/SensorDebugEditorSystemComponent.cpp:42]:
```suggestion
```

**Review by patrykantosz — APPROVED** [review]
LGTM. Just a few, minor suggestions.

**Comment by michalpelka** [comment]
@patrykantosz thanks for review, I applied all suggestions. Since I have no capability to continue work on this PR, could you test if your suggestions are not causing any side effects?

**Comment by patrykantosz** [comment]
With the code changes everything works as expected. Great work, it has really helped me with some other projects.

**Review by pijaro — APPROVED** [review]
I've tested it locally and can confirm it is quite a handy gem. Thanks!

**Comment by michalpelka** [comment]
Thanks for taking care of @pijaro @patrykantosz !

**Pull request merged**